desired_behavior.md

```diff
@@ -0,0 +1,248 @@
+# Scenarios:
+
+## 2.0 CLR / .NET 2.0 FRAMEWORK
+
+### CLR is Present / FRAMEWORK is Present
+
+Behavior:
+
+* Application uses System version of CLR
+* Application uses System version of FRAMEWORK 
+
+### CLR is Present / FRAMEWORK is Not Present
+
+Not Possible
+
+### CLR is Not Present / FRAMEWORK is Not Present
+
+Behavior:
+
+* Application uses Habitat version of CLR
+* Application uses Habitat version of FRAMEWORK 
+
+## 2.0 CLR / .NET 3.0 FRAMEWORK
+
+### CLR is Present / FRAMEWORK is Present
+
+Behavior:
+
+* Application uses System version of CLR
+* Application uses System version of FRAMEWORK 
+
+### CLR is Present / FRAMEWORK is Not Present
+
+Behavior:
+
+* Application uses System version of CLR
+* Application uses Habitat version of FRAMEWORK 
+
+### CLR is Not Present / FRAMEWORK is Not Present
+
+Behavior:
+
+* Application uses Habitat version of CLR
+* Application uses Habitat version of FRAMEWORK 
+
+## 2.0 CLR / .NET 3.5 FRAMEWORK
+
+### CLR is Present / FRAMEWORK is Present
+
+Behavior:
+
+* Application uses System version of CLR
+* Application uses System version of FRAMEWORK 
+
+### CLR is Present / FRAMEWORK is Not Present
+
+Behavior:
+
+* Application uses System version of CLR
+* Application uses Habitat version of FRAMEWORK 
+
+### CLR is Not Present / FRAMEWORK is Not Present
+
+Behavior:
+
+* Application uses Habitat version of CLR
+* Application uses Habitat version of FRAMEWORK 
+
+## 2.0 CLR / .NET 3.5.1 FRAMEWORK
+
+### CLR is Present / FRAMEWORK is Present
+
+Behavior:
+
+* Application uses System version of CLR
+* Application uses System version of FRAMEWORK 
+
+### CLR is Present / FRAMEWORK is Not Present
+
+Behavior:
+
+* Application uses System version of CLR
+* Application uses Habitat version of FRAMEWORK 
+
+### CLR is Not Present / FRAMEWORK is Not Present
+
+Behavior:
+
+* Application uses Habitat version of CLR
+* Application uses Habitat version of FRAMEWORK 
+
+## 4.0 CLR / .NET 4.0 FRAMEWORK
+
+### CLR is Present / FRAMEWORK is Present
+
+Behavior:
+
+* Application uses System version of CLR
+* Application uses System version of FRAMEWORK 
+
+### CLR is Present / FRAMEWORK is Not Present
+
+Not Possible
```

Comment:
I believe this is possible. The 4 CLR can exist with the .net 4.6 which is the default setup on server 2016. In which case we'd use the system CLR and Habitat's framework bits.